### PR TITLE
chore: add a publish job for dev bundles

### PR DESCRIPTION
**Pull request by Racer159**
Adds a publish task to publish dev versions of bundles for testing.

### CODE REVIEW

File: tasks/publish.yaml [hunk 1]
```diff
@@ -1,16 +1,44 @@
...
+        description: Path to the UDS bundle being published
+        default: bundle
+      version:
+        description: The version of the bundle to publish
+        required: true
+    actions:
+      - description: Publish bundles for the amd64 and arm64 architectures
+        cmd: |
+          set -e
+          uds publish ${{ .inputs.path }}/uds-bundle-*-amd64-${{ .inputs.version }}.tar.zst oci://${TARGET_REPO}/bundles --no-progress
```
Comment (zachariahmiller) [inline tasks/publish.yaml:40]: I think im totally fine with this being either way, but in line with making the version an input do you want to do the same with TARGET_REPO?

Comment (zachariahmiller) [reply tasks/publish.yaml:40]: No is a totally valid answer 🤣

Comment (Racer159) [reply tasks/publish.yaml]: So is YES :sweat_smile:

**Comment by Racer159** [comment]
Here is what it would look like to implement this in another repo: https://github.com/defenseunicorns/uds-package-gitlab/pull/76/files

**Review by zachariahmiller — APPROVED** [review]

**Pull request merged**